### ISSUE TITLE
Trojan flagged within project files

### ISSUE THREAD

**Issue by shawncy** [oleg-shilo/wixsharp#287]
Please remove the following files from this project:

wixsharp-master/Source/src/NbsBuilder/NbsBuilder.Src.7z|Output/nbs.exe
wixsharp-master/Source/src/NbsBuilder/NbsBuilder.Src.7z|Output/nbsbuilder.exe
wixsharp-master/Source/src/WixSharp.Samples/nbsbuilder.exe

These are flagged as containing the following Trojan by my antivirus:

Gen:Variant.Razy.240011

I've already reported the user who committed these files to Github.

Thanks!

**Comment by oleg-shilo** [comment]
It does look like false positive.
WinDefender has no problems with it.

The `nbsbuilder.exe` in the archive is identical to the one that is under Git (https://github.com/oleg-shilo/wixsharp/blob/master/Source/src/WixSharp.Samples/nbsbuilder.exe). Thus I don't see how one can be reported as positive and another one is not.

Anyway, in case of any concern please rebuild the executables from the source. This will give you some confidence.

**Comment by shawncy** [comment]
Hi Oleg,

Is that link correct? (https://github.com/oleg-shilo/wixsharp/blob/master/Source/src/WixSharp.Samples/nbsbuilder.exe) If so, this was flagged as containing the Trojan.

Please note that the antivirus (Vipre) is flagging these files:

https://github.com/oleg-shilo/wixsharp/blob/master/Source/src/NbsBuilder/NbsBuilder.Src.7z (contents)
https://github.com/oleg-shilo/wixsharp/blob/master/Source/src/WixSharp.Samples/nbsbuilder.exe

And that these files were recently modified by the following commit:

https://github.com/oleg-shilo/wixsharp/blob/master/Source/src/NbsBuilder/NbsBuilder.Src.7z

From a user with no other activity:

https://github.com/TigranAnchorfree

Can you please double-check?

Thanks,

Shawn

**Comment by oleg-shilo** [comment]
OK, I have temporary replaced the exy under SC with the innocent dummy. 
I had a few conversations with Tigran and he seems to be real. :)
I will rebuild the binary myself and see if any problems there...

**Comment by shawncy** [comment]
Sounds good, Oleg. If you can vouch that Tigran didn't do this intentionally, I'll send Github a note with this (I did report that user to Github).

Thank you so much for the prompt response!

Shawn

**Comment by oleg-shilo** [comment]
It does look a bit confusing.
Tigran, has reported issue as  @geghamyan (#273), where he referred his very own PR (https://github.com/oleg-shilo/wixsharp/pull/272). Though he used a different name (@TigranAnchorfree ) for committing the PR code. That's why his second identity has no activity.

![image](https://user-images.githubusercontent.com/16729806/35891654-89b39b6a-0bf9-11e8-96ad-d0177cddd20b.png)

Anyway, to avoid any further confusion:
- I replaced the `nbsbuilde.exe` in _samples_ with its previous previous (pre PR#272) version.
-  I added the source code for `nbs` and `nbsbuilder`. So the changes are traceable. The source code is also (pre PR#272). Just as it was migrated from CodePlex.
- NbsBuilder.Src.7z has been removed as it is now part of the project codebase
- I have manually patched the `nbsbuilder.cpp` and `Utils.h` with Tigran's changes. Thus codebase now is the latest one.

However I did not bring change s to the project structure (e.g. xp support) as this needs to be discussed between Tigran and me. Tht's why `nbsbuilde.exe` has been effectively rolled back binary wise.

Another reason for the old binaries being placed under GIT is that I cannot build the new `nsbbuilder.exe` right now as I have no working C++ build environment for this.

I will work with Tigran (@geghamyan) on this one.

**Comment by geghamyan** [comment]
Guys, I am sure it's false positive. Oleg probably eventually will rebuild it from the sources, or we can setup it as a part of the whole build.

**Comment by geghamyan** [comment]
Run it through the virustotal - only "MalwarePatrol" out of 67 antiviruses complained. 
https://www.virustotal.com/#/url/20011d3f0e4fe0209bc2160acb804f99db6f95b22bbbcfbdf420ec786c6ae251/detection

**Comment by oleg-shilo** [comment]
Thanks Tigran, 

I have checked and some online checkers also report positives (e.g. 1 out of 67). I am sure it's a random byte combination that makes it to scream. 

Though in these situations I do prefer people to be able to build the exes by themselves. 

Tigran, can you please have a look at the code now and make the VS project buildable. No need to make it part of the release-build. The idea is that one needs to open VS2010-* and be able to build the exe. Possibly with the help of the instructions in the readme.md. The lover version of VS the better.

Right now my VS2017 cannot locate `afxres.h`. Am I missing some include settings?

You can create a dedicated issue for "building nbsbuilder" if feel that is would help.

**Comment by oleg-shilo** [comment]
BTW some of my CS-Script (cscs.exe) versions were also triggering false positeves...

**Comment by geghamyan** [comment]
afxres.h is a part of MFC support, excluded by default component installation.
![image](https://user-images.githubusercontent.com/12634915/35893289-247d8004-0b62-11e8-99a9-f661e8e976d1.png)

**Comment by shawncy** [comment]
Thanks for the help, guys. I'm taking this to Vipre to have them confirm if this is indeed a false positive, and potentially update their AV definition to exclude these flagged files.

I've also confirmed that the most recent version of this repo scans clean in Vipre.

Finally, I'll send out a new note to Github disregard my earlier report to them, Tigran. Hopefully, they get that in time, and that it doesn't cause you any further difficulty.

Thanks again!

**Comment by geghamyan** [comment]
@shawncy, no worries, I hope I won't be banned from the GitHub :)

**Comment by geghamyan** [comment]
@oleg-shilo, why do you want to keep the nbsbuilder on VS2010 ?

**Comment by oleg-shilo** [comment]
Yes, but...

On my old VS2010 it does not build any more. Something has changed. 

VS2017 is ideal for rebuilding but I am afraid of introducing accidental dependencies. Thus in the last project changes a new target runtime was introduced. I was able to build under VS2017 with targeting old vc runtime (idel combination for us) and manually including fully qualified path to `afxres.h`.

I am trying to modify my VS2017 now. Cannot find the VS installer applet :)

If it works then we will go this way. Latest VS with build targeting the older runtime.

**Comment by geghamyan** [comment]
I think as long as you keep the static linking of the CRT - there will be no dependencies.
![image](https://user-images.githubusercontent.com/12634915/35893704-263a37e6-0b64-11e8-849f-878ec44216aa.png)

**Comment by geghamyan** [comment]
There are no external dependencies for VS2017 built binary, except for built-in OS components.
![image](https://user-images.githubusercontent.com/12634915/35894082-f8760496-0b65-11e8-967a-80ad844c3657.png)

**Comment by geghamyan** [comment]
@oleg-shilo ,  VS2017 builds the project when I replace afxres.h with winres.h. I think it effectively removes MFC dependency

**Comment by oleg-shilo** [comment]
Excellent. I also has removed "shlwpi.lib" from the linker input.
Now I only need to remove post-build events and all done.

**Comment by oleg-shilo** [comment]
Done. 6c769e2a4100d5a2bbf4472dc3e5aaf91a162f60

Please note that the current (freshly rebuilt) binaries are still flagged as infected by some antiviruses (1-3/67 on VirusTotal):
[report on nbs.exe](https://www.virustotal.com/#/file/b47c27d24be315d18d0aee1b318f0d08ab3e43ebcad01c99bc0804ea085abad3/detection)
[report on nbsbuilder.exe](https://www.virustotal.com/#/file/aa85e82cd399a1da955c4b7f92a5c39fdc27c4b64d5331bc1d7048c2be664589/detection)

Though now user can rebuild binaries if there are any concerns.

@shawncy please close this issue when you beleive it is completely resolved.

Thank you to everyone who participated :)

**Comment by shawncy** [comment]
I downloaded the latest source, and the raw source was scanned cleanly by Vipre. Once I compiled the solution located at:

https://github.com/oleg-shilo/wixsharp/tree/master/Source/src/NbsBuilder/NbsBuilder.Src/nbsbuilder.sln

Vipre immediately flagged the compiled output as malware, as you indicated, Oleg.

I think you've resolved this issue satisfactorily; I'm going to work with Vipre, using this source code, to try and see if I can get them to update their AV definition to exclude this compiled executable.

Thanks everyone! Issue closed.

By the way, I really appreciate this library. Keep up the good work!

Shawn
